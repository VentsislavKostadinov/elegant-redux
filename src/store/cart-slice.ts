import { createSlice } from '@reduxjs/toolkit'
import { CartState } from './redux-model/CartProps'

const initialState: CartState = {
    items: [],
}

export const cartSlice = createSlice({
    name: 'cart',
    initialState: initialState,
    reducers: {
        addToCart() {},
        removeFromCart() {},
    },
})
