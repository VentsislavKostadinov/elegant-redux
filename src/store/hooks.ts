import { useDispatch, useSelector, TypedUseSelectorHook } from 'react-redux'
import { DispatchFunction } from './redux-model/DispatchFunctionProps'
import { RootState } from './store'

export const useCartDispatch: DispatchFunction = useDispatch

export const useCartSelector: TypedUseSelectorHook<RootState> = useSelector
