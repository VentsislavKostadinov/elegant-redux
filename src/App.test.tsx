import { App } from './App'
import { render } from '@testing-library/react'
import { Provider } from 'react-redux'
import { store } from './store/store'

describe('Render App correctly', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(
            <Provider store={store}>
                <App />
            </Provider>,
        )
        expect(asFragment).toMatchSnapshot()
    })
})
