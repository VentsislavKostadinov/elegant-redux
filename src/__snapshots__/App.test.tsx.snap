// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render App correctly matches snapshot 1`] = `[Function]`;
